### PR TITLE
Add stage name to e-mail subject

### PR DESCRIPTION
**Pull request by brianjmurrell**
Also extend dockerBuildArgs() to do some steps optionally, defaulting to
existing behaviour.

### CODE REVIEW

File: vars/stepResult.groovy [hunk 1]
```diff
@@ -114,7 +114,7 @@ def call(Map config= [:]) {
                  ".  " + comment_url
 
       if (commitPragma(pragma: "Skip-PR-comments") == "true") {
-          emailext subject: "${JOB_NAME} ${config.name} stage ${config.result}",
+          emailext subject: "${config.name} status: ${config.result}",
```
Comment (phender) [inline vars/stepResult.groovy:117]: This doesn't seem to align with the PR description.  Isn't this removing the job name from the email subject?

**Comment by ashleypittman** [comment]
On reflection I'm not sure what emails this relates to.  All mine seem to come from Github, aside from the occasional master failed to build one.

File: vars/stepResult.groovy (hunk 1 above)
Comment (brianjmurrell) [reply vars/stepResult.groovy:117]: Ick.  Yeah.  Let me take another look at this.  It must just be the one other part that needs to be in this PR.

**Comment by brianjmurrell** [comment]
> On reflection I'm not sure what emails this relates to.

This refers to the e-mails that Jenkins will send directly, when it doesn't add comments (which are noise in my opinion) to PRs when you add the `Skip-PR-comments: true` commit pragma to your commit.

https://github.com/daos-stack/pipeline-lib/blob/af4fdedcf49288a09022ec83c7cf04ff4113e912/vars/stepResult.groovy#L116-L135

> All mine seem to come from Github, aside from the occasional master failed to build one.

Yes, without `Skip-PR-comments: true` that is the current behaviour.  This is a new (optional, default disabled) behaviour I am testing out to see how much less noisy PRs can be when we don't spam it with comments about every failed build/test.  Ultimately, I find those useful at the time of the failure, but being left behind in the history of the PR is making our PRs messy and noisy, and sometimes unrenderable (a pink unicorn) in GitHub.

**Review by jolivier23 — APPROVED** [review]

**Comment by jolivier23** [comment]
Looks like it needs to merge latest master

**Pull request merged**